packages/navie/src/services/openai-completion-service.ts

```diff
@@ -181,11 +181,17 @@ export default class OpenAICompletionService implements CompletionService {
     public readonly modelName: string,
     public readonly temperature: number,
     private trajectory: Trajectory,
-    private readonly messageTokenReducerService: MessageTokenReducerService
+    private readonly messageTokenReducerService: MessageTokenReducerService,
+    private apiUrl?: string,
+    private apiKey?: string
   ) {
     this.model = new ChatOpenAI({
       modelName: this.modelName,
       temperature: this.temperature,
+      configuration: {
+        baseURL: this.apiUrl ?? process.env.OPENAI_BASE_URL,
+      },
+      apiKey: this.apiKey ?? process.env.OPENAI_API_KEY,
```

Comment:
Maybe leave just `this.apiUrl`/`this.apiKey` and add a comment that if they're undefined langchain will automatically use values from the environment if present? As it is, it puzzled me for a moment to consider what would happen in the Azure case (it will in fact work correctly, at least I think so).